tools/differ/README.md

```diff
@@ -0,0 +1,23 @@
+# Dataset Differ
+
+This utility generates a diff (point and series analysis) of two data files of the same dataset for import analysis.
+
+**Usage**
+```
+python differ.py --currentData=<filepath> --previousData=<filepath>
+```
+
+Parameter description
+currentDataFile: Location of the current MCF data file
+previousDataFile: Location of the previous MCF data file
+groupbyColumns: Columns to group data for diff analysis in the order var,place,time etc. Default value: “variableMeasured,observationAbout,observationDate”
+valueColumns: Columns with statvar value (unit etc.) for diff analysis. Default value:  "value,unit"
```

Comment:
Wondering if unit should be a group-by column.
For instance, if there are 2 SVObs for Temp in Celsius and Fahrenheit,
Prev:Max_Temp,PlaceDcid1,2024,80,F
Prev:Max_Temp,PlaceDcid1,2024,26.2,C

Cur:Max_Temp,PlaceDcid1,2024,27,C
Would this be compared to (Prev:Max_Temp,PlaceDcid1,2024,26.2,C) or (Prev:Max_Temp,PlaceDcid1,2024,80,F)?

Reply:
I am not sure. If we add it as a group by column, it will treat a change in the unit as a new series. If we add it to value columns, it will treat it as a modified data point. I think the latter makes more sense.

Reply:
If an import has same data in multiple units, if the unit in value can compare the right units, then this is fine:
say: previous has 
Max_Temp,PlaceDcid1,2024,80,F
Max_Temp,PlaceDcid1,2024,26.2,C

current has:
Max_Temp,PlaceDcid1,2024,26.21,C
Max_Temp,PlaceDcid1,2024,80,F

Only the first in current should modified. It shouldn;t compare F with C due to the change in output ordering.
If the current version does that we can go ahead.

Reply:
We concatenate value+unit to create the target value which is then compared. Having two different value points for the same sv_ob is not supported in the implementation.